sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataGrpcMultiplexer2.java

```diff
@@ -253,15 +253,13 @@ public void onError(Throwable t) {
           "Failed to handle for {}",
           apiServiceDescriptor == null ? "unknown endpoint" : apiServiceDescriptor,
           t);
-      outboundObserver.onCompleted();
     }
 
     @Override
     public void onCompleted() {
       LOG.warn(
           "Hanged up for {}.",
           apiServiceDescriptor == null ? "unknown endpoint" : apiServiceDescriptor);
-      outboundObserver.onCompleted();
     }
   }
 }
```

Comment:
Maybe we can rename this class to just BeamFnDataGrpcMultiplexer with a follow up?

Reply:
Yeah, I wanted to do that in a separate CL since the commit history would be easier to understand.

Reply:
https://github.com/apache/beam/pull/25144 is the follow-up